client/js/verification-portal/do-not-list/action-creators.js

```diff
@@ -0,0 +1,30 @@
+import { DO_NOT_LIST } from './action-types'
+import { store } from '../../hah-app/index'
+
+export async function updateDoNotList(doNotList) {
+	const response = await fetch(
+		`/api/pregnancy-centers/${store.getState().resource._id}/do-not-list`,
+		{
+			method: 'PUT',
+			credentials: 'include',
+			headers: {
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify({ doNotList: doNotList }),
+		},
+	)
+	const jsonResponse = await response.json()
+
+	store.dispatch(updateDoNotListActionCreator(jsonResponse.doNotList))
+
+	return jsonResponse
+}
+
+export function updateDoNotListActionCreator(doNotList) {
+	doNotList = doNotList ? doNotList : false
```

Comment:
@bhavyaab It's a best practice to _not_ reassign a function parameter. You can force `doNotList` to be a true/false value by coercing the value in the response on line 14.
Ex: `body: JSON.stringify({ doNotList: !!doNotList })`